Web/PathPieces.hs

```diff
@@ -122,6 +125,24 @@ instance PathPiece a => PathMultiPiece [a] where
     fromPathMultiPiece = mapM fromPathPiece
     toPathMultiPiece = map toPathPiece
 
+-- | Since 0.2.1. A function for helping generate free 'PathPiece'
+--   instances for enumeration data types 
+--   that have derived 'Read' and 'Show' instances.
+--   Intended to be used like this:
+--
+--   > data MyData = Foo | Bar | Baz
+--   >   deriving (Read,Show)
+--   > instance PathPiece MyData where
+--   >   fromPathPiece = readFromPathPiece
+--   >   toPathPiece = showToPathPiece
+--
+readFromPathPiece :: Read s => S.Text -> Maybe s
+readFromPathPiece = readMaybe . S.unpack
+
+-- | Since 0.2.1. See the documentation for 'readFromPathPiece'.
```

Comment:
Not a huge deal, but the general convention is to have the "Since" part as the last line, with a newline between it and the documentation https://github.com/yesodweb/yesod/blob/d8a831162c3a7387b3ac47dfbdbdff20fc211ee4/yesod-form/Yesod/Form/Fields.hs#L352-L354